src/Compilers/CSharp/Portable/Lowering/IteratorRewriter/IteratorMethodToStateMachineRewriter.cs

```diff
@@ -386,8 +388,11 @@ public override BoundNode VisitTryStatement(BoundTryStatement node)
             //      return;
             // }
             Debug.Assert(frame.parent.finalizeState == _currentFinallyFrame.finalizeState);
-            rewrittenHandler = F.Block(
+            rewrittenHandler = F.Block((object)this.cachedThis != null?
+                                            ImmutableArray.Create(this.cachedThis):
```

Comment:
The extension would be on  ```T: class``` . It feels like it would be very noisy.

---
In reply to: [87635783](https://github.com/dotnet/roslyn/pull/14959#discussion_r87635783) [](ancestors = 87635783)